docs/machine-learning/tutorials/movie-recommmendation.md

```diff
@@ -0,0 +1,479 @@
+
+E.g. if User 1 and User 2 rate movies similarly, then User 2 is more likely to enjoy a movie that User 1 has watched and rated highly:
+
+|       | Incredibles 2 (2018) | The Avengers (2012) | Guardians of the Galaxy (2014) |
+| -------------:|-------------:| -----:|-----:|
+| User 1 | Watched and liked movie | Watched and liked movie | Watched and liked movie |
+| User 2 | Watched and liked movie | Watched and liked movie | Has not watched -- RECOMMEND movie |
+
+The `Matrix Factorization` trainer has several [Options](xref:Microsoft.ML.Trainers.MatrixFactorizationTrainer.Options) including:
+
+**Input and output columns**
+* <xref:Microsoft.ML.Trainers.MatrixFactorizationTrainer.Options.MatrixColumnIndexColumnName> (`userIdEncoded`)
+* <xref:Microsoft.ML.Trainers.MatrixFactorizationTrainer.Options.MatrixRowIndexColumnName> (`movieIdEncoded`)
+* <xref:Microsoft.ML.Trainers.MatrixFactorizationTrainer.Options.LabelColumnName> (`Label`)
+
+**[Hyperparameters](../resources/glossary.md#hyperparameter)**
+
+* <xref:Microsoft.ML.Trainers.MatrixFactorizationTrainer.Options.NumberOfIterations> 
+* <xref:Microsoft.ML.Trainers.MatrixFactorizationTrainer.Options.ApproximationRank> 
+
+For more information on hyperparameters, see the [Improve your model](#improve-your-model) step below.
+
+
+Add the following as the next line of code in the `BuildAndTrainModel()` method:
+
+<!--[!code-csharp[FitModel](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#FitModel "Call the Fit method and return back the trained model")]-->
+
+```csharp
+Console.WriteLine("=============== Training the model ===============");
+model = est.Fit(trainingDataView);
+
+return model;
+```
+
+The [Fit()](xref:Microsoft.ML.Trainers.MatrixFactorizationTrainer.Fit%28Microsoft.Data.DataView.IDataView,Microsoft.Data.DataView.IDataView%29) method trains your model with the provided training dataset. Technically, it executes the `Estimator` definitions by transforming the data and applying the training, and it returns back the trained model, which is a `Transformer`.
+
+Add the following as the next line of code in the `Main()` method to call your `BuildAndTrainModel()` method and return the trained model:
+
+[!code-csharp[BuildTrainModelMain](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#BuildTrainModelMain "Add BuildAndTrainModel method in Main")]
+
+## Evaluate your model
+
+Once you have trained your model, use your test data to evaluate how your model is performing. 
+
+Outside of `Main()`, create a method called `EvaluateModel()`:
+```csharp
+public static void EvaluateModel(MLContext mlcontext, IDataView testDataView, ITransformer model)
+{
+
+
+}
+```
+
+Define the method by adding the following code to `EvaluateModel()`:
```

Comment:
Lines 276 - 312 need to have the new snippets added from https://github.com/dotnet/samples/pull/703, and explanations before each snippet.